.github/workflows/build-and-test.yml

```diff
@@ -21,19 +21,43 @@ jobs:
     uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
     with:
       runsonlabels: '["macOS", "self-hosted"]'
-      scheme: SpeziSpeech
-      artifactname: SpeziSpeech.xcresult
+      scheme: SpeziSpeech-Package
+      xcodeversion: 15.0
+      artifactname: SpeziSpeech-Package.xcresult
   ios:
     name: Build and Test iOS
     uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
     with:
       runsonlabels: '["macOS", "self-hosted"]'
       path: 'Tests/UITests'
       scheme: TestApp
+      xcodeversion: 15.0
```

Comment:
I think we can omit the version here, Xcode 15 or newer should be the default selection in our setup.